libs/@local/hash-backend-utils/src/pg-tables.ts

```diff
@@ -2,74 +2,106 @@
-      entityTypeVersionId: row.entity_type_version_id as string,
-      // eslint-disable-next-line @typescript-eslint/no-unsafe-assignment
-      properties:
-        typeof row.properties === "string"
-          ? JSON.parse(row.properties)
-          : row.properties,
-      createdBy: row.created_by as string,
-    };
+export const entityEditionFromWalJsonMsg = (
+  msg: Wal2JsonMsg<SupportedRealtimeTable>,
+): EntityEdition => {
+  if (msg.table !== "entity_editions") {
+    throw new Error(
+      `invalid table "${msg.table}" for 'entity_edition' parsing`,
+    );
   }
+  const obj = Object.fromEntries(
+    msg.columns.map(({ name, value }) => [name, value]),
+  );
 
-  static parseWal2JsonMsg(msg: Wal2JsonMsg): Entity {
-    if (msg.table !== "entities") {
-      throw new Error(`invalid table "${msg.table}" for an 'entities' update`);
-    }
-    const obj = Object.fromEntries(
-      msg.columns.map(({ name, value }) => [name, value]),
+  return {
+    entity_edition_id: obj.entity_edition_id as string,
+    properties: obj.properties as Record<string, unknown>,
+    left_to_right_order: obj.left_to_right_order as number,
+    right_to_left_order: obj.right_to_left_order as number,
+    record_created_by_id: obj.record_created_by_id as string,
+    archived: obj.archived as boolean,
+  };
+};
+
+export type EntityTemporalMetadata = {
+  owned_by_id: string;
+  entity_uuid: Record<string, unknown>;
+  entity_edition_id: string;
+  decision_time: string;
+  transaction_time: string;
+};
+
+export const entityTemporalMetadataFromWalJsonMsg = (
+  msg: Wal2JsonMsg<SupportedRealtimeTable>,
+): EntityTemporalMetadata => {
+  if (msg.table !== "entity_temporal_metadata") {
+    throw new Error(
+      `invalid table "${msg.table}" for 'entity_temporal_metadata' parsing`,
     );
-    return this.parseFromRow(obj);
   }
-}
+  const obj = Object.fromEntries(
+    msg.columns.map(({ name, value }) => [name, value]),
+  );
 
-export class Link {
-  constructor(
-    /**
-     * @todo Update table definition when provenance info is added for updates
-     *   https://app.asana.com/0/1202805690238892/1202848989198291/f
-     */
-    public sourceEntityId: string,
-    public targetEntityId: string,
-    public linkTypeVersionId: string,
-    public createdBy: string,
-  ) {}
+  return {
+    owned_by_id: obj.owned_by_id as string,
+    entity_uuid: obj.entity_uuid as Record<string, unknown>,
+    entity_edition_id: obj.entity_edition_id as string,
+    decision_time: obj.decision_time as string,
+    transaction_time: obj.transaction_time as string,
+  };
+};
 
-  private static parseFromRow(row: Record<string, unknown>): Link {
-    return {
-      sourceEntityId: row.source_entity_id as string,
-      targetEntityId: row.target_entity_id as string,
-      linkTypeVersionId: row.link_type_version_id as string,
-      createdBy: row.created_by as string,
-    };
-  }
+export type Entity = EntityEdition & EntityTemporalMetadata;
```

Comment:
Do we perhaps want to name all of these types something else to avoid yet another competing `Entity` definition?

Reply:
(And why can we not just use the normal `Entity`?)

Reply:
Yeah, I'll prefix with `Pg` i think. We can't reuse `Entity` because we're not receiving the `Entity`, we're trying to reconstruct this from the DB change stream, so only a subset of the metadata is available